File: lab 7/java/src/Table.java
public class Table
{
    private  int currentlyEating;
    private int size;
    public boolean waits = false;
    Table(int size)
    {
        this.currentlyEating = 0;
        this.size = size;
    }

    public boolean canStartEating()
    {
        return (currentlyEating < size);
    }

    public void startEating()
    {
        currentlyEating++;
    }

    public void finishEating()
    {
        currentlyEating--;
    }
}

File: lab 7/scala/src/main/scala/Table.scala
class Table(var size: Int) {
  this.currentlyEating = 0
  private var currentlyEating = 0
  var waits = false

  def canStartEating: Boolean = currentlyEating < size

  def startEating(): Unit = {
    currentlyEating += 1
  }

  def finishEating(): Unit = {
    currentlyEating -= 1
  }
}

